lib/hexo/validate_config.js

```diff
@@ -9,8 +9,11 @@ module.exports = ctx => {
   if (typeof config.url !== 'string') {
     throw new TypeError(`Invalid config detected: "url" should be string, not ${typeof config.url}!`);
   }
-  if (config.url.trim().length <= 0) {
-    throw new TypeError('Invalid config detected: "url" should not be empty!');
+  try {
+    // eslint-disable-next-line no-new
+    new URL(config.url);
+  } catch {
+    throw new TypeError('Invalid config detected: "url" should starts with "http://" or "https://"');
```

Comment:
```suggestion
    throw new TypeError('Invalid config detected: "url" should be a valid URL');
```

IMHO, `starts with "http://" or "https://"` should only be used in docs, since any invalid url would trigger the `TypeError`.

Reply:
I agree `should be a valid URL` is more universal, since `http://localhost:123456` will also throw error.